staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -950,16 +951,25 @@ func (e *Store) updateForGracefulDeletionAndFinalizers(ctx context.Context, name
 }
 
 // Delete removes the item from storage.
-func (e *Store) Delete(ctx context.Context, name string, options *metav1.DeleteOptions) (runtime.Object, bool, error) {
+func (e *Store) Delete(ctx context.Context, name string, deleteValidation rest.ValidateObjectFunc, options *metav1.DeleteOptions) (runtime.Object, bool, error) {
 	key, err := e.KeyFunc(ctx, name)
 	if err != nil {
 		return nil, false, err
 	}
 	obj := e.NewFunc()
 	qualifiedResource := e.qualifiedResourceFromContext(ctx)
-	if err := e.Storage.Get(ctx, key, "", obj, false); err != nil {
+	err = e.Storage.Get(ctx, key, "", obj, false)
+	if err == nil || errors.IsNotFound(err) {
+		// Note that we continue the admission check on "Not Found" error is for the compatibility
```

Comment:
@liggitt highlighting the comment here😃FYI

Reply:
@yue9944882 do you recall why you left this comment? I checked the noderestriction admission code and still didn't understand.